packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -32,6 +34,25 @@ const { trackBuildError, decorateEvent } = require(`gatsby-telemetry`)
 import errorParser from "./api-runner-error-parser"
 import { wrapNode, wrapNodes } from "./detect-node-mutations"
 
+// Override createContentDigest to remove autogenerated data from nodes to
+// ensure consistent digests.
+function createContentDigest(node) {
```

Comment:
I have worries about this (for cases when it's not used for actual nodes). Do you think it would be reasonable to have some heuristic trying to determine if object looks like gatsby's node before sanitizing it?

For example check like this and early bail if it's not satisfied?
```suggestion
function createContentDigest(node) {
  if (!node?.internal?.type) {
    // this doesn't look like a node, so let's just pass it as-is
    return _createContentDigest(node)
  }
```

Reply:
great idea 👍